### PR TITLE
update spring-boot-integration.md for [add AiServiceRegisteredEvent]

### PR DESCRIPTION
**Pull request by catofdestruction**
PR: https://github.com/langchain4j/langchain4j-spring/pull/89

## Issue
original PR link: https://github.com/langchain4j/langchain4j-spring/pull/77
Issues link: langchain4j/langchain4j#2112

## Change
update spring-boot-integration.md for [add AiServiceRegisteredEvent]

### CODE REVIEW

File: docs/docs/tutorials/spring-boot-integration.md [hunk 1]
```diff
@@ -187,6 +187,35 @@ In this case, you must explicitly specify **all** components.
 
 More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
 
+### Listening for AI Service Registration Events
+
+After you have completed the development of the AI Service in a declarative manner, you can also listen for the
+`AiServiceRegisteredEvent` by implementing the `ApplicationListener<AiServiceRegisteredEvent>` interface.
+
+This event is triggered when AI services are registered in the Spring context, 
+allowing you to obtain information about all registered AI services and their tools at runtime.
```
Comment (kpavlov) [inline docs/docs/tutorials/spring-boot-integration.md:195]: Not related to the PR: I am just thinking out loud, is AiServiceRegisteredEvent necessary it is possible to get all beans by class from ApplicationContext `Map<String, T> getBeansOfType(Class<T> type)`

**Review by kpavlov — DISMISSED** [review]

File: docs/docs/tutorials/spring-boot-integration.md (hunk 1 above)
Comment (catofdestruction) [reply docs/docs/tutorials/spring-boot-integration.md:195]: > Not related to the PR: I am just thinking out loud, is AiServiceRegisteredEvent necessary it is possible to get all beans by class from ApplicationContext `Map<String, T> getBeansOfType(Class<T> type)`

This method can obtain the class information of the AI service, but cannot retrieve more details, such as the description of the tools. (`List<ToolSpecification> toolSpecifications`)

File: docs/docs/tutorials/spring-boot-integration.md [hunk 2]
```diff
@@ -187,6 +187,35 @@ In this case, you must explicitly specify **all** components.
...
+`AiServiceRegisteredEvent` by implementing the `ApplicationListener<AiServiceRegisteredEvent>` interface.
+
+This event is triggered when AI services are registered in the Spring context, 
+allowing you to obtain information about all registered AI services and their tools at runtime. 
+This provides the application with dynamic configuration, debugging, and extension capabilities,
+making it particularly suitable for scenarios that require specific actions based on service registration status.
+Here is an example:
+```java
+@SpringBootApplication
+class AiServiceWithToolsApplication implements ApplicationListener<AiServiceRegisteredEvent> {
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]: nit: `ApplicationListener` should probably not be a `@SpringBootApplication`, perharps make it a regular component?

**Review by ghost — COMMENTED** [review]
@catofdestruction thank you!

File: docs/docs/tutorials/spring-boot-integration.md (hunk 2 above)
Comment (catofdestruction) [reply docs/docs/tutorials/spring-boot-integration.md]: done

File: docs/docs/tutorials/spring-boot-integration.md [hunk 3]
```diff
@@ -187,6 +187,31 @@ In this case, you must explicitly specify **all** components.
...
+`AiServiceRegisteredEvent` by implementing the `ApplicationListener<AiServiceRegisteredEvent>` interface.
+
+This event is triggered when AI services are registered in the Spring context, 
+allowing you to obtain information about all registered AI services and their tools at runtime. 
+This provides the application with dynamic configuration, debugging, and extension capabilities,
+making it particularly suitable for scenarios that require specific actions based on service registration status.
+Here is an example:
+```java
+@Component
+class AiServiceRegistrar implements ApplicationListener<AiServiceRegisteredEvent> {
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion
class AiServiceRegisteredEventListener implements ApplicationListener<AiServiceRegisteredEvent> {

```

File: docs/docs/tutorials/spring-boot-integration.md [hunk 4]
```diff
@@ -187,6 +187,31 @@ In this case, you must explicitly specify **all** components.
 
 More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
 
+### Listening for AI Service Registration Events
+
+After you have completed the development of the AI Service in a declarative manner, you can also listen for the
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion
After you have completed the development of the AI Service in a declarative manner, you can listen for the

```

File: docs/docs/tutorials/spring-boot-integration.md [hunk 5]
```diff
@@ -187,6 +187,31 @@ In this case, you must explicitly specify **all** components.
 
 More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
 
+### Listening for AI Service Registration Events
+
+After you have completed the development of the AI Service in a declarative manner, you can also listen for the
+`AiServiceRegisteredEvent` by implementing the `ApplicationListener<AiServiceRegisteredEvent>` interface.
+
+This event is triggered when AI services are registered in the Spring context,
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion
This event is triggered when AI Service is registered in the Spring context, 

```

File: docs/docs/tutorials/spring-boot-integration.md [hunk 6]
```diff
@@ -187,6 +187,31 @@ In this case, you must explicitly specify **all** components.
...
 More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
 
+### Listening for AI Service Registration Events
+
+After you have completed the development of the AI Service in a declarative manner, you can also listen for the
+`AiServiceRegisteredEvent` by implementing the `ApplicationListener<AiServiceRegisteredEvent>` interface.
+
+This event is triggered when AI services are registered in the Spring context, 
+allowing you to obtain information about all registered AI services and their tools at runtime. 
+This provides the application with dynamic configuration, debugging, and extension capabilities,
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion
```

File: docs/docs/tutorials/spring-boot-integration.md [hunk 7]
```diff
@@ -187,6 +187,31 @@ In this case, you must explicitly specify **all** components.
...
+making it particularly suitable for scenarios that require specific actions based on service registration status.
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion
```

**Review by ghost — DISMISSED** [review]
@catofdestruction thank you!

File: docs/docs/tutorials/spring-boot-integration.md [hunk 8]
```diff
@@ -187,6 +187,32 @@ In this case, you must explicitly specify **all** components.
 
 More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
 
+### Listening for AI Service Registration Events
+
+After you have completed the development of the AI Service in a declarative manner, you can listen for the
+
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion

```

File: docs/docs/tutorials/spring-boot-integration.md [hunk 9]
```diff
@@ -187,6 +187,32 @@ In this case, you must explicitly specify **all** components.
 
 More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
 
+### Listening for AI Service Registration Events
+
+After you have completed the development of the AI Service in a declarative manner, you can listen for the
+
+`AiServiceRegisteredEvent` by implementing the `ApplicationListener<AiServiceRegisteredEvent>` interface.
+
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion

```

File: docs/docs/tutorials/spring-boot-integration.md [hunk 10]
```diff
@@ -187,6 +187,32 @@ In this case, you must explicitly specify **all** components.
...
 More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
 
+### Listening for AI Service Registration Events
+
+After you have completed the development of the AI Service in a declarative manner, you can listen for the
+
+`AiServiceRegisteredEvent` by implementing the `ApplicationListener<AiServiceRegisteredEvent>` interface.
+
+This event is triggered when AI Service is registered in the Spring context, 
+
```
Comment (langchain4j) [inline docs/docs/tutorials/spring-boot-integration.md]:
```suggestion

```

**Pull request merged**